packages/report-server/src/reportBuilder/transform/parser/TransformParser.ts

```diff
@@ -65,14 +77,22 @@ export class TransformParser extends ExpressionParser {
       });
       this.set('where', this.whereFunction); // no '@' prefix for where
     }
+
+    this.context = context;
   }
 
-  public evaluate(expression: string) {
-    if (TransformParser.isExpression(expression)) {
-      return super.evaluate(expression.substring(1));
-    }
+  public isExpression(input: unknown) {
+    return typeof input === 'string' && input.startsWith(this.EXPRESSION_PREFIX);
+  }
 
-    return expression;
+  protected readExpression(input: unknown) {
+    return this.isExpression(input)
+      ? (input as string).replace(new RegExp(`^${this.EXPRESSION_PREFIX}`), '')
+      : input;
+  }
+
+  public evaluate(input: unknown) {
+    return this.isExpression(input) ? super.evaluate(input) : input;
```

Comment:
The condition retains the existing functionality, which just returns the input (and avoids validation + evaluation) if the input does not start with `=`